pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -288,7 +285,7 @@ public void deleteSegmentFile() {
   private final SegmentCommitterFactory _segmentCommitterFactory;
   private final ConsumptionRateLimiter _rateLimiter;
 
-  private volatile StreamPartitionMsgOffset _latestStreamOffsetAtStartupTime = null;
+  private final StreamPartitionMsgOffset _latestStreamOffsetAtStartupTime;
```

Comment:
Is this variable not being modified anywhere?

Reply:
nope. It is set once during startup of the `LLRealtimeSegmentDataManager`. After that, it is only read from.